File: apkEditor/src/main/java/com/mcal/apkeditor/dialogs/AppAgreementDialog.java
package com.mcal.apkeditor.dialogs;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.mcal.apkeditor.MainActivity;
import com.mcal.apkeditor.R;
import com.mcal.apkeditor.view.ViewDialog;

import java.lang.ref.WeakReference;

public class AppAgreementDialog {
    private final WeakReference<MainActivity> activityRef;

    @SuppressLint("SetTextI18n")
    public AppAgreementDialog(MainActivity activity) {

        this.activityRef = new WeakReference<>(activity);

        LayoutInflater inflater = LayoutInflater.from(activity);
        View layout = inflater.inflate(R.layout.dlg_app_license, null);
        TextView tv = (TextView) layout.findViewById(R.id.tv_content);
        tv.setText("Information in this dialog is provided in connection with APK Editor. No license, express or implied, by estoppel or otherwise, to any intellectual property rights is granted by this.\n" +
                "\n" +
                "APK Editor is designed for Android fans who know what exactly they are doing, but not intended for hack, please use it under following terms:\n" +
                "\n" +
                "1) Please only modify the apk files which you have intellectual property rights.\n" +
                "\n" +
                "2) For apk files you don't have intellectual property rights, you need to ask for authorities from the developer to modify it. And even though you have rights to modify it, you still need to ask for re-distribution rights to publish it.\n" +
                "\n" +
                "3) To prevent abuse of APK Editor, sign feature is not provided any more.\n" +
                "\n" +
                "4) We may make changes to specifications and product descriptions at any time, without notice.");

        EditText inputEt = (EditText) layout.findViewById(R.id.et_input);

        ViewDialog dialog = new ViewDialog(activity);
        dialog.setTitle("Agreement");
        dialog.setView(layout);
        dialog.setCancelable(false);
        dialog.setPositive(android.R.string.ok, view -> {
            String input = inputEt.getText().toString();
            if (input.trim().toLowerCase().equals("accept")) {
                activityRef.get().initFileWithPermissionCheck();

                SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(activityRef.get());
                SharedPreferences.Editor e = sp.edit();
                e.putBoolean("app_agreement_accepted", true);
                e.apply();

                dialog.dismiss();
            } else {
                Toast.makeText(activityRef.get(), R.string.input_agree_toast, Toast.LENGTH_SHORT).show();
                activityRef.get().finish();
            }
        });
        dialog.setNegative(android.R.string.cancel, view -> {
            activityRef.get().finish();
        });
        dialog.show();
    }

    public static boolean appLicenseAccepted(Context ctx) {
        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(ctx);
        return sp.getBoolean("app_agreement_accepted", false);
    }
}

File: apkEditor/src/main/java/com/mcal/apkeditor/dialogs/AppAgreementDialog.kt
package com.mcal.apkeditor.dialogs

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import androidx.preference.PreferenceManager
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.textfield.TextInputEditText
import com.mcal.apkeditor.R
import com.mcal.apkeditor.activities.MainActivity
import java.util.*

class AppAgreementDialog @SuppressLint("SetTextI18n") constructor(activity: MainActivity) {

    companion object {
        @JvmStatic
        fun appLicenseAccepted(ctx: Context?): Boolean {
            val sp: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(ctx!!)
            return sp.getBoolean("app_agreement_accepted", false)
        }
    }

    init {
        val layout = LayoutInflater.from(activity).inflate(R.layout.dialog_app_license, null)
        val textView = layout.findViewById<TextView>(R.id.tv_content)
        textView.text = if (Locale.getDefault().language.contains("ru")) buildString {
            append("Данная информация относится к приложению ApkEditor Pro.")
            append("\n\n")
            append("APK Editor Pro ApkEditor - приложение для анализа мобильных приложений с возможностью вносить правки в коде, а также тестировать приложение на уязвимости. Он не предназначен для взлома приложений! Пожалуйста используйте его на следующих условиях:")
            append("\n\n")
            append("1) Пожалуйста, изменяйте только те файлы APK, на которые у вас есть права интеллектуальной собственности.")
            append("\n\n")
            append("2) Для APK файлов, на которые у вас нет прав интеллектуальной собственности, вам необходимо запросить полномочия у разработчика, чтобы изменить их. И хотя у вас есть права на его изменение, вам все равно нужно запрашивать права на повторное распространение для его публикации.")
            append("\n\n")
            append("3) Мы можем вносить изменения в спецификации и описания продуктов в любое время без предварительного уведомления.")
            append("\n\n")
            append("Если вы не согласны с выше перечисленными пунктами, пожалуйста удалите приложение и не используйте его.")
        } else buildString {
            append("This information applies to the ApkEditor Pro application.")
            append("\n\n")
            append("APK Editor Pro ApkEditor is an application for analyzing mobile applications with the ability to make changes in the code, as well as test the application for vulnerabilities. It is not designed to hack apps! Please use it under the following conditions:")
            append("\n\n")
            append("1) Please only modify APK files for which you have intellectual property rights.")
            append("\n\n")
            append("2) For APK files for which you do not have intellectual property rights, you need to request permission from the developer to change them. And while you have rights to modify it, you still need to request redistribution rights to publish it.")
            append("\n\n")
            append("3) We may make changes to product specifications and descriptions at any time without notice.")
            append("\n\n")
            append("If you do not agree with the above points, please delete the application and do not use it.")
        }

        val textInputView = layout.findViewById<TextInputEditText>(R.id.et_input)
        val dialog = MaterialAlertDialogBuilder(activity)
        dialog.setTitle(activity.getString(R.string.agreement))
        dialog.setView(layout)
        dialog.setCancelable(false)
        dialog.setPositiveButton(android.R.string.ok) { v, _ ->
            val input = textInputView.text.toString()
            if (input.trim().lowercase(Locale.getDefault()) == "accept") {
                activity.initFileWithPermissionCheck()
                PreferenceManager.getDefaultSharedPreferences(activity).edit().putBoolean("app_agreement_accepted", true).apply()
                v.dismiss()
            } else {
                Toast.makeText(activity, R.string.input_agree_toast, Toast.LENGTH_SHORT).show()
                activity.finish()
            }
        }
        dialog.setNegativeButton(android.R.string.cancel) { _, _ -> activity.finish() }
        dialog.show()
    }
}